diff -ruN a/bfd/config.bfd b/bfd/config.bfd
--- a/bfd/config.bfd	2017-07-10 06:54:41.000000000 -0300
+++ b/bfd/config.bfd	2018-11-11 21:25:07.078246185 -0200
@@ -754,6 +754,11 @@
     targ_selvecs=
     targ64_selvecs=x86_64_elf64_vec
     ;;
+  i[3-7]86-*-chicago*)
+    targ_defvec=i386_elf32_vec
+    targ_selfvecs=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
 #ifdef BFD64
   x86_64-*-cloudabi*)
     targ_defvec=x86_64_elf64_cloudabi_vec
diff -ruN a/config.sub b/config.sub
--- a/config.sub	2017-07-10 06:54:41.000000000 -0300
+++ b/config.sub	2017-12-22 17:20:23.000000000 -0200
@@ -1417,7 +1417,8 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* \
+		  | -chicago*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ruN a/gas/configure.tgt b/gas/configure.tgt
--- a/gas/configure.tgt	2017-07-10 06:54:41.000000000 -0300
+++ b/gas/configure.tgt	2017-12-22 17:31:05.000000000 -0200
@@ -291,6 +291,7 @@
   i386-*-chaos)				fmt=elf ;;
   i386-*-rdos*)				fmt=elf ;;
   i386-*-darwin*)			fmt=macho ;;
+  i386-*-chicago*)			fmt=elf ;;
 
   i860-*-*)				fmt=elf endian=little ;;
 
diff -ruN a/ld/configure.tgt b/ld/configure.tgt
--- a/ld/configure.tgt	2017-07-10 06:54:42.000000000 -0300
+++ b/ld/configure.tgt	2018-11-11 21:25:52.547823348 -0200
@@ -405,6 +405,9 @@
 			targ_extra_libpath=$targ_extra_emuls
 			tdir_elf_i386_nacl=`echo ${targ_alias} | sed -e 's/x86_64/i386/'`
 			;;
+i[3-7]86-*-chicago*)	targ_emul=elf_i386_chicago
+			targ_extra_emuls=elf_i386
+			;;
 i860-*-coff)		targ_emul=coff_i860 ;;
 i860-stardent-sysv4* | i860-stardent-elf*)
 			targ_emul=elf32_i860
@@ -937,6 +940,10 @@
   NATIVE_LIB_DIRS='/usr/local/lib /usr/ccs/lib /lib /usr/lib'
   ;;
 
+*-*-chicago*)
+  NATIVE_LIB_DIRS='/Development/Libraries'
+  ;;
+
 esac
 
 case "${target}" in
diff -ruN a/ld/emulparams/elf_i386_chicago.sh b/ld/emulparams/elf_i386_chicago.sh
--- a/ld/emulparams/elf_i386_chicago.sh	1969-12-31 21:00:00.000000000 -0300
+++ b/ld/emulparams/elf_i386_chicago.sh	2018-11-11 00:44:15.762490731 -0200
@@ -0,0 +1,4 @@
+. ${srcdir}/emulparams/elf_i386.sh
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
+ENTRY=AppEntry
diff -ruN a/ld/Makefile.am b/ld/Makefile.am
--- a/ld/Makefile.am	2017-07-10 06:54:42.000000000 -0300
+++ b/ld/Makefile.am	2018-11-11 21:26:44.317321755 -0200
@@ -288,6 +288,7 @@
 	eelf32xstormy16.c \
 	eelf32xtensa.c \
 	eelf_i386.c \
+	eelf_i386_chicago.c \
 	eelf_i386_be.c \
 	eelf_i386_chaos.c \
 	eelf_i386_fbsd.c \
@@ -1344,6 +1345,10 @@
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_i386_chicago.c: $(srcdir)/emulparams/elf_i386_chicago.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_chicago "$(tdir_elf_i386_chicago)"
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
diff -ruN a/ld/Makefile.in b/ld/Makefile.in
--- a/ld/Makefile.in	2017-09-15 08:38:08.000000000 -0300
+++ b/ld/Makefile.in	2018-11-11 21:27:20.436960125 -0200
@@ -657,6 +657,7 @@
 	eelf32xstormy16.c \
 	eelf32xtensa.c \
 	eelf_i386.c \
+	eelf_i386_chicago.c \
 	eelf_i386_be.c \
 	eelf_i386_chaos.c \
 	eelf_i386_fbsd.c \
@@ -1320,6 +1321,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_be.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_chaos.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_chicago.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_fbsd.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_ldso.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/eelf_i386_nacl.Po@am__quote@
@@ -2908,6 +2910,10 @@
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_i386_chicago.c: $(srcdir)/emulparams/elf_i386_chicago.sh \
+  $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+	${GENSCRIPTS} elf_i386_chicago "$(tdir_elf_i386_chicago)"
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
