diff -ruN a/config.sub b/config.sub
--- a/config.sub	2016-12-29 09:04:53.721993000 -0200
+++ b/config.sub	2018-11-11 00:47:47.960806811 -0200
@@ -1409,7 +1409,8 @@
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* \
+        | -chicago*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -ruN a/fixincludes/mkfixinc.sh b/fixincludes/mkfixinc.sh
--- a/fixincludes/mkfixinc.sh	2016-06-21 18:57:20.986088000 -0300
+++ b/fixincludes/mkfixinc.sh	2018-11-11 00:53:46.145027450 -0200
@@ -19,7 +19,8 @@
     powerpc-*-rtems*   | \
     powerpcle-*-eabisim* | \
     powerpcle-*-eabi* | \
-    *-musl* )
+    *-musl* | \
+    *-chicago*)
 	#  IF there is no include fixing,
 	#  THEN create a no-op fixer and exit
 	(echo "#! /bin/sh" ; echo "exit 0" ) > ${target}
diff -ruN a/gcc/config/chicago.h b/gcc/config/chicago.h
--- a/gcc/config/chicago.h	1969-12-31 21:00:00.000000000 -0300
+++ b/gcc/config/chicago.h	2018-11-11 21:22:23.732936010 -0200
@@ -0,0 +1,31 @@
+#undef TARGET_CHICAGO
+#define TARGET_CHICAGO 1
+
+#undef STANDARD_STARTFILE_PREFIX
+#define STANDARD_STARTFILE_PREFIX "/Development/Libraries/"
+
+#undef LIB_SPEC
+#define LIB_SPEC "-lc" /* link everything against the C standard library */
+
+#undef LIBSTDCXX
+#define LIBSTDCXX "cxx" /* CHicago's c++ library is called libcxx (the gcc's default is called libstdc++) */
+
+/* Files that are linked before user code.
+   The %s tells GCC to look for these files in the library directory. */
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared:crt0.o%s} crti.o%s %{!shared:crtbegin.o%s}"
+
+/* Files that are linked after user code. */
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "%{!shared:crtend.o%s} crtn.o%s"
+
+#undef LINK_SPEC
+#define LINK_SPEC "%{shared:-shared} -z max-page-size=4096"
+
+/* Don't automatically add extern "C" { } around header files. */
+#undef  NO_IMPLICIT_EXTERN_C
+#define NO_IMPLICIT_EXTERN_C 1
+
+/* Additional predefined macros. */
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS() do { builtin_define ("__chicago__"); builtin_assert ("system=chicago"); } while(0);
\ Nenhum caractere de nova linha no final do arquivo
diff -ruN a/gcc/config.gcc b/gcc/config.gcc
--- a/gcc/config.gcc	2017-07-31 08:33:18.182361000 -0300
+++ b/gcc/config.gcc	2018-11-11 21:27:51.116645999 -0200
@@ -915,6 +915,12 @@
   default_use_cxa_atexit=yes
   use_gcc_stdint=wrap
   ;;
+*-*-chicago*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  native_system_header_dir=/Development/Headers
+  ;;
 esac
 
 case ${target} in
@@ -1795,6 +1801,9 @@
 	tmake_file="${tmake_file} i386/t-x86_64-elf"
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h fuchsia.h"
 	;;
+i[34567]86-*-chicago*)
+    tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h chicago.h"
+    ;;
 ia64*-*-elf*)
 	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h ia64/sysv4.h ia64/elf.h"
 	tmake_file="ia64/t-ia64"
diff -ruN a/libgcc/config.host b/libgcc/config.host
--- a/libgcc/config.host	2017-05-10 17:42:58.366374000 -0300
+++ b/libgcc/config.host	2018-11-11 21:28:14.499735853 -0200
@@ -757,6 +757,10 @@
 		extra_parts="$extra_parts vtv_start.o vtv_end.o vtv_start_preinit.o vtv_end_preinit.o"
 	fi
 	;;
+i[34567]86-*-chicago*)
+	extra_parts="$extra_parts crti.o crtbegin.o crtend.o crtn.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-crtstuff-pic t-libgcc-pic"
+	;;
 ia64*-*-elf*)
 	extra_parts="$extra_parts crtbeginS.o crtendS.o crtfastmath.o"
 	tmake_file="ia64/t-ia64 ia64/t-ia64-elf ia64/t-eh-ia64 t-crtfm t-softfp-tf ia64/t-softfp t-softfp ia64/t-softfp-compat"
